base/REPLCompletions.jl

```diff
@@ -381,12 +381,51 @@ function bslash_completions(string, pos)
     return (false, (String[], 0:-1, false))
 end
 
+function dict_identifier_key(str,tag)
+    if tag === :string
+        str_close = str*"\""
+    elseif tag === :cmd
+        str_close = str*"`"
+    else
+        str_close = str
+    end
+
+    frange, end_of_indentifier = find_start_brace(str_close, c_start='[', c_end=']')
+    isempty(frange) && return (nothing, nothing, nothing)
+    identifier = Symbol(str[frange[1]:end_of_indentifier])
+    isdefined(Main,identifier) || return (nothing, nothing, nothing)
+    begin_of_key = findnext(x->!in(x,whitespace_chars), str, end_of_indentifier+2)
+    begin_of_key==0 && return (identifier, nothing, nothing)
+    partial_key = str[begin_of_key:end]
+    main_id = eval(Main,identifier)
+    typeof(main_id) <: Associative && length(main_id)<1e6 || return (identifier, nothing, nothing)
+    identifier, partial_key, begin_of_key
```

Comment:
Should this return `main_id` so you only have to call `eval` once? (See the call for `keys($identifier)` below.)